### PR TITLE
Add tests for patch feature in temor-script

### PR DESCRIPTION
**Pull request by diru1100**
# Pull request

## Description
- Added tests covering various operations under the patch feature.

## Related

* Related Issues: contributes to #721 
* [Patch specification](https://datatracker.ietf.org/doc/html/rfc6902)

## Checklist

* [x] The RFC, if required, has been submitted and approved
* [x] Any user-facing impact of the changes is reflected in docs.tremor.rs
* [x] The code is tested
* [x] Use of unsafe code is reasoned about in a comment
* [x] Update CHANGELOG.md appropriately, recording any changes, bug fixes, or other observable changes in behavior
* [x] The performance impact of the change is measured (see below)

### CODE REVIEW

File: tremor-script/eqc/spec.erl [hunk 1]
```diff
@@ -164,6 +162,44 @@ spec_uop_int(S, N) ->
...
+	    [spec_inner_float(S, N - 1)]).
+
+patch_operation(S, N) ->
+    frequency([{1,
+		{insert, spec_inner_string(S, N - 1),
+		 spec_inner_no_float(S, N - 1)}},
+	       {1,
+		{upsert, spec_inner_string(S, N - 1),
+		 spec_inner_no_float(S, N - 1)}},
+	       {1,
```
Comment (Licenser) [inline tremor-script/eqc/spec.erl:179]: `patch_operation` does not cover all specified patch operations

File: tremor-script/eqc/spec.erl [hunk 2]
```diff
@@ -164,6 +162,44 @@ spec_uop_int(S, N) ->
...
+		{merge, spec_inner_string(S, N - 1),
+		 spec_inner_record(S, N - 1)}},
+	       {1, {erase, spec_inner_string(S, N - 1)}}]).
+
+% Test cases the function spec_uop_record generates
```
Comment (Licenser) [inline tremor-script/eqc/spec.erl:184]: this documentaiton is in the wrong place

File: tremor-script/eqc/model.erl [hunk 3]
```diff
@@ -35,6 +32,56 @@ combine_values(Key, SpecVal = #{}, Acc) ->
...
+patch_operation({upsert, Key, Value}, Acc) ->
+    % does what we expect from upsert
+    maps:put(Key, Value, Acc);
+patch_operation({erase, Key}, Acc) ->
+    maps:remove(Key, Acc).
+
+-spec ast_eval(#vars{}, {}) -> {#vars{},
+				integer() | float() | boolean() | binary()}.
+
+% Test cases the patch eval function generates
```
Comment (Licenser) [inline tremor-script/eqc/model.erl:48]: this documentation is in the wrong place

**Review by Licenser — CHANGES_REQUESTED** [review]
There are a few errors in this

File: tremor-script/eqc/model.erl (hunk 3 above)
Comment (diru1100) [reply tremor-script/eqc/model.erl]: Yeah, I kept it near patch_operation code

File: tremor-script/eqc/spec.erl (hunk 2 above)
Comment (diru1100) [reply tremor-script/eqc/spec.erl]: Updated, thanks

File: tremor-script/eqc/spec.erl (hunk 1 above)
Comment (diru1100) [reply tremor-script/eqc/spec.erl:189]: Yeah, I missed some 😅. Updated and ran tests again, thanks

**Review by Licenser — COMMENTED** [review]
Weeeh! :rocket:

**Review by mfelsche — APPROVED** [review]
LGTM, but I wasnt able to test it locally and CI also didnt run it. The next CI run on main will tell :)

**Review by Licenser — APPROVED** [review]
```
===== test_eqc:prop_simple_expr
****************************************************************************************************(x10)******************************************************************************************(x1)*
OK, passed 1001 tests

===== test_eqc:prop_simple_expr_with_state
****************************************************************************************************(x10)******************************************************************************************(x1)*
OK, passed 1001 tests
```

**Pull request merged**